addons/web/models/models.py

```diff
@@ -877,9 +877,15 @@ def onchange(self, values: dict, field_names: list[str], fields_spec: dict):
             missing_names = [fname for fname in fields_spec if fname not in values]
             defaults = self.default_get(missing_names)
             for field_name in missing_names:
-                values[field_name] = defaults.get(field_name, False)
                 if field_name in defaults:
+                    values[field_name] = defaults[field_name]
                     field_names.append(field_name)
+                else:
+                    field = self._fields[field_name]
+                    if not field.compute or self.pool.field_depends[field]:
```

Comment:
It is maybe out of scope and can be done latter :thinking:  
IMO, we should target all compute fields, not only the ones that have no depends. The ones with depends was invalidated anyway by the modified/onchange mechanism, then it doesn't make lot of sense to force it to False for them, right ?

Reply:
I'll put that on the "for later" stack :smile: